stdlib/auth/src/main/ballerina/src/auth/inbound_basic_auth_provider.bal

```diff
@@ -36,12 +36,12 @@ public type InboundBasicAuthProvider object {
 
     *InboundAuthProvider;
 
-    public BasicAuthConfig basicAuthConfig;
+    public readonly BasicAuthConfig basicAuthConfig;
```

Comment:
Thinking out loud, as to what approach we should follow by default.

I think we should make `InboundBasicAuthProvider` a `readonly object` instead. Reasons being the type definition will mandate only valid possibly immutable types are allowed for fields, and developers won't introduce non-readonly fields by mistake.

Reply:
Yes, `readonly object` is better and cleaner. Will push changes.